mlshvch/hometask-4/my_hash.rb

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'digest'
+# class MyHash with custom implementation of Hash Table
+class MyHash
+  include Enumerable
+  EMPTY_HASH_TABLE_SIZE = 10
+
+  def initialize
+    create_empty_hash
+  end
+
+  def create_empty_hash(hash_table_size = EMPTY_HASH_TABLE_SIZE)
+    @hash_table_size = hash_table_size
+    @hash_table = Array.new(EMPTY_HASH_TABLE_SIZE, [])
+    @number_of_entries = 0.0
+    self
+  end
+
+  def calculate_hash_table_index(key)
+    # sha = Digest::SHA2.hexdigest(key.to_s)
+    # hash = ''
+    # 10.times do |index|
+    #   hash += sha[index].ord.to_s
+    # end
+    # hash.to_i % @hash_table_size
+    key.hash % @hash_table_size
+  end
+
+  def []=(key, value)
+    index = calculate_hash_table_index(key)
+    @hash_table[index] = if @hash_table[index].to_a.empty? || @hash_table[index].to_a.assoc(key)
```

Comment:
> if @hash_table[index].to_a.empty? || @hash_table[index].to_a.assoc(key)

Why do you need to check for emptiness first? You will get the same outcome with find / assoc anyway.

> assoc(key)

Nice, I did not remember about such method myself.